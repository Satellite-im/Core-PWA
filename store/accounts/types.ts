export enum RegistrationStatus {
  UKNOWN = 'unknown',
  IN_PROGRESS = 'in_progress',
  FUNDING_ACCOUNT = 'funding_account',
  SENDING_TRANSACTION = 'sending_transaction',
  REGISTERED = 'registered',
}

export interface AccountsState {
  loading?: boolean
  locked: boolean
  pin?: string
  pinHash: string
  active: string
  gasPrice: string
  phrase: string
  error: string
  encryptedPhrase: string
  registered: boolean
  registrationStatus: RegistrationStatus
}

export enum AccountsError {
  INVALID_PIN = 'errors.accounts.invalid_pin',
  PIN_TOO_SHORT = 'errors.accounts.pin_too_short',
  UNABLE_TO_CREATE_MNEMONIC = 'errors.accounts.unable_to_create_mnemonic',
  MNEMONIC_NOT_PRESENT = 'errors.accounts.mnemonic_not_present',
  USER_DERIVATION_FAILED = 'errors.accounts.user_derivation_failed',
  USER_NOT_REGISTERED = 'errors.accounts.user_not_registered',
  USER_ALREADY_REGISTERED = 'errors.accounts.user_already_registered',
  PAYER_NOT_PRESENT = 'errors.accounts.payer_not_present',
}

export interface UserRegistrationPayload {
  name: string
  photoHash: string
  status: string
}
