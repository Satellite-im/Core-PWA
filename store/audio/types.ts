export interface AudioState {
  previousVolume: number
  inputVolume: number
  muted: boolean
  volume: number
  deafened: boolean
  sounds: {
    inboundMedia: number
    outboundMedia: number
    system: number
  }
}
