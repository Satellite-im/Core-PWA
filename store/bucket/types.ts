export interface BucketState {
  initialized: boolean
}
