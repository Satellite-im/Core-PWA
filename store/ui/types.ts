import { TranslateResult } from 'vue-i18n'
import { FileMessage } from '~/types/textile/mailbox'
import { Glyph } from '~/types/ui/glyph'
import { Channel } from '~/types/ui/server'
import { Alert } from '~/libraries/ui/Alerts'
import { ConversationMessage } from '~/libraries/Iridium/chat/types'

export enum GlyphMarketViewStatus {
  HOME = 'home',
  SHOP_ALL = 'shop_all',
  SHOP_DETAIL = 'shop_detail',
}

export enum ModalWindows {
  NEW_FOLDER = 'newfolder',
  CREATE_SERVER = 'createServer',
  MARKETPLACE = 'marketplace',
  WALLET = 'wallet',
  WALLET_MINI = 'walletMini',
  ERROR = 'error',
  CHANGELOG = 'changelog',
  GLYPH = 'glyph',
  USER_PROFILE = 'userProfile',
  CALL_TO_ACTION = 'callToAction',
  RENAME_FILE = 'renameFile',
}

export interface EnhancerInfo {
  show: boolean
  floating?: boolean
  position?: number[]
  defaultWidth?: string
  defaultHeight?: string
  containerWidth?: number
  route: string
}

export interface EmojiUsage {
  code: string
  count: number
  content: string
}

export interface RecentGlyph {
  pack: Glyph
  url: string
  count: number
}

export enum SettingsRoutes {
  PERSONALIZE = 'personalize',
  PROFILE = 'profile',
  AUDIO_AND_VIDEO = 'audio & Video',
  KEY_BINDS = 'keybinds',
  ACCOUNTS_AND_DEVICES = 'accounts & Devices',
  PRIVACY = 'privacy',
  DEVELOPER = 'developer',
  INFO = 'info',
  NOTIFICATIONS = 'notifications',
  STORAGE = 'storage',
  NETWORK = 'network',
  REALMS = 'realms',
}

export type ContextMenuItemTypes = 'primary' | 'danger'

export interface ContextMenuItem {
  text: string | TranslateResult
  func: Function
  type?: ContextMenuItemTypes
}

export interface UIState {
  contextMenuStatus: boolean
  contextMenuValues: ContextMenuItem[]
  quickProfile: object | boolean
  userProfile: object
  notifications: Alert[]
  contextMenuPosition: object
  quickProfilePosition: object
  showSettings: boolean
  settingsSideBar: boolean
  settingsRoute: SettingsRoutes
  showSidebarUsers: boolean
  showSearchResult: boolean
  showSidebar: boolean
  modals: {
    [key in ModalWindows]: boolean | object
  }
  glyphModalPack: string
  chatbarContent: string
  chatbarFocus: boolean
  replyChatbarMessage?: ConversationMessage
  showPinned: boolean
  fullscreen: boolean
  enhancers: EnhancerInfo
  messages: any[]
  unreadMessage: number
  isScrollOver: boolean
  showOlderMessagesInfo: boolean
  isTyping: object | boolean
  isReacted: boolean
  activeChannel: Channel | undefined
  settingReaction: object
  hoveredGlyphInfo: object | undefined
  glyphMarketplaceView: object
  editMessage: {
    id: string
    from: string
    payload: string
  }
  mostEmojiUsed: EmojiUsage[]
  recentGlyphs: RecentGlyph[]
  chatImageOverlay?: FileMessage
  isMobileNavVisible: boolean
}

export type Position = {
  x: number
  y: number
}
