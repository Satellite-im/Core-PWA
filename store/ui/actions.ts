import Mousetrap from 'mousetrap'
import Vue from 'vue'
import { v4 as uuidv4 } from 'uuid'
import { Position, UIState } from './types'
import SoundManager, { Sounds } from '~/libraries/SoundManager/SoundManager'
import TextileManager from '~/libraries/Textile/TextileManager'
import { ActionsArguments } from '~/types/store/store'
import { Friend } from '~/types/ui/friends'
import { Channel } from '~/types/ui/server'
import { getFullUserInfoFromState } from '~/utilities/Messaging'
import { getCorrectKeybind } from '~/utilities/Keybinds'
import { TextileError } from '~/store/textile/types'
import { AlertType } from '~/libraries/ui/Alerts'

const $Sounds = new SoundManager()

export default {
  setMessages({ commit }: ActionsArguments<UIState>, messages: any[]) {
    commit('setMessages', messages)
  },
  sendMessage({ commit, rootState }: ActionsArguments<UIState>, message: any) {
    if (message.user.address !== rootState.accounts.active) {
      $Sounds.playSound(Sounds.NEW_MESSAGE)
    }
    commit('sendMessage', message)
  },
  setIsScrollOver({ commit }: ActionsArguments<UIState>, status: boolean) {
    commit('setIsScrollOver', status)
  },
  setIsReacted({ commit }: ActionsArguments<UIState>, status: boolean) {
    commit('setIsReacted', status)
  },
  setActiveChannel({ commit }: ActionsArguments<UIState>, channel: Channel) {
    commit('setActiveChannel', channel)
  },
  /**
   * @method openSettings
   * @description Opens setting page
   * @example Mousetrap.bind('ctrl+s', dispatch('audio/toggleMute') )
   */
  openSettings({ commit, state }: any) {
    commit('toggleSettings', { show: !state.showSettings })
  },
  /**
   * @method activateKeybinds
   * @description Activates all keybindings with Mousetrap
   * @example mounted (){ activateKeybinds() }
   */
  async activateKeybinds({ dispatch, rootState }: ActionsArguments<UIState>) {
    const { toggleMute, toggleDeafen, openSettings, callActiveChat } =
      // @ts-ignore
      rootState.settings.keybinds
    Mousetrap.reset()
    Mousetrap.bind(getCorrectKeybind(toggleMute), (event: KeyboardEvent) => {
      event.preventDefault()
      dispatch('audio/toggleMute', null, { root: true })
    })
    Mousetrap.bind(getCorrectKeybind(toggleDeafen), (event: KeyboardEvent) => {
      event.preventDefault()
      dispatch('audio/toggleDeafen', null, { root: true })
    })
    Mousetrap.bind(getCorrectKeybind(openSettings), (event: KeyboardEvent) => {
      event.preventDefault()
      dispatch('openSettings')
    })
    Mousetrap.bind(
      getCorrectKeybind(callActiveChat),
      (event: KeyboardEvent) => {
        event.preventDefault()
        dispatch('webrtc/call', { kinds: ['audio'] }, { root: true })
      },
    )
  },
  /**
   * @method setNotifications
   * @description Collects all existing notifications for a user
   */
  async setNotifications({ commit }: ActionsArguments<UIState>) {
    const $TextileManager: TextileManager = Vue.prototype.$TextileManager

    if (!$TextileManager.notificationManager?.isInitialized()) {
      throw new Error(TextileError.MAILBOX_MANAGER_NOT_INITIALIZED)
    }
    const notifications =
      await $TextileManager.notificationManager?.getnotifications()
    commit('setNotifications', notifications)
  },
  async sendNotification(
    { commit, rootState }: ActionsArguments<UIState>,
    payload: {
      message: string
      from: string
      imageHash: string
      title: string
      type: AlertType
    },
  ) {
    const $TextileManager: TextileManager = Vue.prototype.$TextileManager

    if (!$TextileManager.notificationManager?.isInitialized()) {
      throw new Error(TextileError.MAILBOX_MANAGER_NOT_INITIALIZED)
    }
    const notificationResponse =
      await $TextileManager.notificationManager?.sendNotification({
        from: payload.from,
        id: uuidv4(),
        title: payload.title,
        imageHash: payload.imageHash,
        message: payload.message,
        type: payload.type,
      })
    commit('sendNotification', notificationResponse)
  },
  removeSeenNotification(
    { commit, rootState }: ActionsArguments<UIState>,
    notificationId: string,
  ) {
    commit('notificationSeen', notificationId)
  },
  /**
   * @method clearKeybinds
   * @description Unbinds all current keybindings with Mousetrap
   * @example destroyed (){ clearKeybinds() }
   */
  async clearKeybinds({ dispatch }: ActionsArguments<UIState>) {
    Mousetrap.reset()
  },
  async setChatbarFocus({ dispatch }: ActionsArguments<UIState>) {
    await dispatch('toggleChatbarFocus', false)
    dispatch('toggleChatbarFocus', true)
  },
  toggleChatbarFocus({ commit }: ActionsArguments<UIState>, flag: boolean) {
    commit('setChatbarFocus', flag)
  },
  setChatbarContent(
    { commit, dispatch }: ActionsArguments<UIState>,
    val: {
      content: string
      userId?: string
    },
  ) {
    commit('chatbarContent', val.content)
    if (val.userId)
      dispatch(
        'chat/setChatText',
        { value: val.content, userId: val.userId },
        { root: true },
      )
  },
  showQuickProfile(
    { commit, state, rootState }: ActionsArguments<UIState>,
    payload: { textilePublicKey: string; position: Position },
  ) {
    if (!payload?.textilePublicKey || !payload?.position) {
      return
    }

    const selectedUser = getFullUserInfoFromState(
      payload.textilePublicKey,
      rootState,
    )

    commit('setQuickProfilePosition', payload.position)
    commit('quickProfile', selectedUser)
  },
  async showProfile(
    { commit, rootState, dispatch }: ActionsArguments<UIState>,
    user: Friend,
  ) {
    if (!user) {
      return
    }
    let metadata = null
    if (!user.metadata) {
      const $TextileManager: TextileManager = Vue.prototype.$TextileManager
      metadata = await $TextileManager.metadataManager?.getFriendMetadata(
        user.address,
      )
    }
    commit('toggleModal', {
      name: 'userProfile',
      state: true,
    })
    const friend = {
      ...user,
    }
    if (metadata) {
      friend.metadata = metadata
      commit('friends/updateFriend', friend, { root: true })
    }
    commit('setUserProfile', friend)
  },
}
