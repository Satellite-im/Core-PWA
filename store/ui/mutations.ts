import { NuxtState } from '@nuxt/types/app'

export default {
  toggleContextMenu(state: NuxtState, enabled: Boolean) {
    state.ui = {
      ...state.ui,
      contextMenuStatus: enabled,
    }
  },
  showSidebarUsers(state: NuxtState, enabled: Boolean) {
    state.ui = {
      ...state.ui,
      showSidebarUsers: enabled,
    }
  },
  setContextMenuValues(state: NuxtState, values: any) {
    state.ui = {
      ...state.ui,
      contextMenuValues: values,
    }
  },
  setContextMenuPosition(state: NuxtState, e: any) {
    state.ui = {
      ...state.ui,
      contextMenuPosition: { x: e.x, y: e.y },
    }
  },
  setQuickProfilePosition(state: NuxtState, e: any) {
    state.ui = {
      ...state.ui,
      quickProfilePosition: { x: e.x, y: e.y },
    }
  },
  quickProfile(state: NuxtState, profile: Object | Boolean) {
    state.ui = {
      ...state.ui,
      quickProfile: profile,
    }
  },
  chatbarContent(state: NuxtState, content: String) {
    state.ui = {
      ...state.ui,
      chatbarContent: content,
    }
  },
  fullscreen(state: NuxtState, fullscreen: Boolean) {
    state.ui = {
      ...state.ui,
      fullscreen,
    }
  },
  toggleEnhancers(state: NuxtState, show: Boolean) {
    state.ui = {
      ...state.ui,
      showEnhancers: show,
    }
  },
}
