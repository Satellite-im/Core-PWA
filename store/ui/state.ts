interface UIState {
  contextMenuStatus: Boolean
  contextMenuValues: Array<Object>
  quickProfile: Object | Boolean
  contextMenuPosition: Object
  quickProfilePosition: Object
  showSettings: Boolean
  showSidebarUsers: Boolean
  showSearchResult: Boolean
  modals: Object
  chatbarContent: String
  replyChatbarContent: {
    id: String
    from: String
    payload: String
  }
  fullscreen: Boolean
  showEnhancers: Boolean
  messages: any[]
  unreadMessage: number
  isScrollOver: boolean
  isTyping: Object | Boolean
}

const InitalUIState = (): UIState => ({
  contextMenuStatus: false,
  showSidebarUsers: true,
  showSearchResult: false,
  showSettings: false,
  quickProfile: false,
  contextMenuValues: [],
  contextMenuPosition: { x: 0, y: 0 },
  quickProfilePosition: { x: 0, y: 0 },
  modals: {
    newfolder: false,
    createServer: false,
  },
  chatbarContent: '',
  replyChatbarContent: { id: '', from: '', payload: '' },
  fullscreen: false,
  showEnhancers: false,
  messages: [],
  unreadMessage: 0,
  isScrollOver: false,
  isTyping: false,
})

export default InitalUIState
