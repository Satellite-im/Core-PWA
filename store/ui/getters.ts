import { EmojiUsage, UIState } from './types'
import { Alert, AlertState } from '~/libraries/ui/Alerts'

const getters = {
  getSortedMostUsedEmojis: (state: UIState): EmojiUsage[] => {
    return [...state.mostEmojiUsed].sort((a, b) => b.count - a.count)
  },
  getSortedRecentGlyphs: (state: UIState) => {
    return [...state.recentGlyphs].sort((a, b) => b.count - a.count)
  },
  showSidebar: (state: UIState) => {
    return state.showSidebar
  },
  allUnseenNotifications: (state: UIState): Alert[] => {
    return [...state.notifications]
      .sort((a, b) => {
        return b.at - a.at
      })
      .filter((noti) => {
        return noti.state === AlertState.UNREAD
      })
  },
  getAlertCount: (state: UIState) => {
    return state.notificationsAlertCounter
  },
  swiperSlideIndex: (state: UIState) => {
    return state.swiperSlideIndex
  },
  isFilesIndexLoading: (state: UIState): boolean => {
    return Boolean(state.filesUploadStatus)
  },
}

export default getters
