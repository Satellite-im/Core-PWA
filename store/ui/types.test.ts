import { expect } from '@jest/globals'
import { Flairs, Themes } from '~/store/ui/types'

describe('check flair constants', () => {
  const inst: any = Flairs

  it('should return the flair constants', () => {
    expect(inst).toMatchSnapshot()
  })

  it('should not return the flair constants', () => {
    expect(inst).not.toEqual({})
  })
})

describe('check theme constants', () => {
  const inst: any = Themes

  it('should return the theme constants', () => {
    expect(inst).toMatchSnapshot()
  })

  it('should not return the theme constants', () => {
    expect(inst).not.toEqual({})
  })
})
