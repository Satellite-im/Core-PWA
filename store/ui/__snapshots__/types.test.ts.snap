// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check flair constants should return the flair constants 1`] = `
Array [
  Object {
    "text": "Satellite",
    "value": Array [
      "#2761fd",
      "#286CFE",
      "39, 97, 253",
    ],
  },
  Object {
    "text": "Peach",
    "value": Array [
      "#ED4C67",
      "#ED5672",
      "237, 76, 103",
    ],
  },
  Object {
    "text": "Pink",
    "value": Array [
      "#FDA7DF",
      "#FDB1E9",
      "253, 167, 223",
    ],
  },
  Object {
    "text": "Lime",
    "value": Array [
      "#A3CB38",
      "#AED542",
      "163, 203, 56",
    ],
  },
  Object {
    "text": "Purple",
    "value": Array [
      "#6F1E51",
      "#80215D",
      "111, 30, 81",
    ],
  },
  Object {
    "text": "Lavender",
    "value": Array [
      "#9980FA",
      "#A891FF",
      "153, 128, 250",
    ],
  },
  Object {
    "text": "Sunflower",
    "value": Array [
      "#FFC312",
      "#FACA3E",
      "255, 195, 18",
    ],
  },
  Object {
    "text": "Deep",
    "value": Array [
      "#30336b",
      "#2D328A",
      "48, 51, 107",
    ],
  },
  Object {
    "text": "Void",
    "value": Array [
      "#2C3A47",
      "#36434F",
      "44, 58, 71",
    ],
  },
]
`;

exports[`check theme constants should return the theme constants 1`] = `
Array [
  Object {
    "class": "",
    "label": "Default",
    "value": "default",
  },
  Object {
    "class": "moonless_night",
    "label": "Moonless Night",
    "value": "moonless_night",
  },
]
`;
